uwblueprint/sistema
# teacher profile unstyled

# Notion ticket link



[Implement Unstyled Teacher Profile Page](https://www.notion.so/uwblueprintexecs/Implement-Unstyled-Teacher-Profile-Page-e416bc3eb4ab495784014d3e36f0cfbf?pvs=4)



## Implementation description

- add profile.tsx page with user info, absences, sign out
- update Dockerfile for prisma schema
- add endpoint to fetch user info based on email from gauth (userid and num of absences)
- add endpoint to fetch user absences given user id
- update .env to replace localhost with db (??) so `db:5432` instead

![image](https://github.com/uwblueprint/sistema/assets/91142355/eba275ab-c4cc-4359-a366-c281721918a9)



## Steps to test

1. run docker
2. ensure db is seeded (see PR #12 and #19 for details)
3. log in with google on localhost:3000
4. go to localhost:3000/profile and validate info is correct
5. go to console to ensure the email logged is correct (code has a fake email from the seeded db)



## What should reviewers focus on?

- Dockerfile changes
- whether profile.tsx has React best practices

## Checklist
